### PR TITLE
Clean up code and refactor

### PR DESCRIPTION
**Pull request by basnijholt**
Still a work in progress. This is an attempt to simplify the code.

Edit: done.

### CODE REVIEW

**Comment by ohld** [comment]
LGTM but with one question about deleted file

**Comment by sudoguy** [comment]
Need to fix issues from Travis CI. After, I'll review code

**Comment by sudoguy** [comment]
Maybe we should use [Hue](https://github.com/UltimateHackers/hue) instead of termcolor?
Hue works on Windows terminals, support bright colors and bold/italic strings.
What do you think, @basnijholt ?

**Comment by ohld** [comment]
Still need to fix some Travis codestyle issues. Anyway good job!

**Comment by basnijholt** [comment]
> Maybe we should use Hue instead of termcolor?
Hue works on Windows terminals, support bright colors and bold/italic strings.
What do you think, @basnijholt ?

I agree that it would be good. I just made the change in d7dd670b3847728a9876e2bc17dd44d396010898.

> Still need to fix some Travis codestyle issues.

To be honest, I am following PEP8 where it's failing now...

**Comment by basnijholt** [comment]
I rewrote that function, all tests should pass now :)

**Review by ohld — DISMISSED** [review]
You guys made a great job! Your commits are awesome!

**Comment by sudoguy** [comment]
@basnijholt We have some problem with that refactoring. Some methods have the same name. ```get_user_followers``` in example. Or ```get_user_followings```. 
If we look at Method Resolution Order of class Bot, we will see that:
```(instabot.bot.bot.Bot, instabot.api.api.API, object)```
When you use get_user_followings, then python get method from class Bot.
And that broke ```get_total_followers_or_followings```. 
Try to run example ```unfollow_nonfollowers.py```

**Comment by sudoguy** [comment]
Your work is really awesome! But we need to solve these problems.

**Comment by ohld** [comment]
Your pull request is definitely going to be the new 0.3.6 version!

**Comment by ohld** [comment]
@basnijholt Do you think that all Instant architecture is really bad? 
If you have some time please check [Instapro](https://github.com/instagrambot/instapro) repository. 
There I introduced the way to make Instagram queries in parallel between several Instagram accounts. 
For example, I can get medias by hashtag in parallel, but still I need to put likes on them from the main account. 

Also there I make User class with all his data. I dump all his data to the disk and load from the disk to prevent multiple logins -> lot prevent bans.

**Comment by basnijholt** [comment]
So I think I am done for now! 😄 

There are many more changes I want to make but I think we should put it in a different PR.

As you can see I changed all the variable names to PEP-8 compliant names and changed the style at a lot of places.

Also, I unified some functions while retaining all functionality. 

Furthermore, I made the `Bot` have an attribute `api` instead of subclassing it. I believe this is the right abstraction. We should still have compatibility because I added a bunch of properties in the `Bot` class.

**Comment by basnijholt** [comment]
Now I am really done!

Sorry that it contains so many changes, but splitting this up into many separate PRs is going to take me almost the same amount of time as it took me to get here. I understand this will be a difficult task to review. Please forgive me :)

I have some more idea on how to massively simplify the way `instabot` handles files, after we will be able to remove a few 100 lines of code, but I'll wait for this to be merged first!

Then when the code is in good order we can think about how to correctly handle the multi user case.

File: instabot/bot/bot_archive.py [hunk 1]
```diff
@@ -7,11 +7,11 @@ def archive(self, media_id, undo=False):
     delay.small_delay(self)
     media = self.get_media_info(media_id)
     media = media[0] if isinstance(media, list) else media
-    if super(self.__class__, self).archiveMedia(media, undo):
+    if self.api.archive_media(media, undo):
         self.total_archived += int(not undo)
         self.total_unarchived += int(undo)
         return True
-    self.logger.info("Media with %s is not %s ." % media_id, 'unarchived' if undo else 'archived')
+    self.logger.info("Media id %s is not %s .", media_id, 'unarchived' if undo else 'archived')
```
Comment (basnijholt) [inline instabot/bot/bot_archive.py]: Are you sure this is correct? I think it should be 
```python
"Media id %s is not %s ." % (media_id, 'unarchived' if undo else 'archived')
```

Comment (sudoguy) [reply instabot/bot/bot_archive.py]: It works on both variants. See [PEP 282](https://www.python.org/dev/peps/pep-0282/). It's recommended this style. Old PEP, old style, yes...
More readable article [here](https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html)

**Review by sudoguy — APPROVED** [review]
Nice job!

**Pull request merged**